IdentityCore/src/util/MSIDKeyOperationUtil.m

```diff
@@ -0,0 +1,130 @@
+//
+//  MSIDKeyOperationUtil.m
+//  IdentityCore
+//
+//  Created by Ameya Patil on 9/24/21.
+//  Copyright © 2021 Microsoft. All rights reserved.
+//
+
+#import "MSIDKeyOperationUtil.h"
+#import "MSIDJwtAlgorithm.h"
+#import "NSData+JWT.h"
+#import "NSData+MSIDExtensions.h"
+
+@implementation MSIDKeyOperationUtil
+
++ (MSIDKeyOperationUtil *)sharedInstance
+{
+    static dispatch_once_t once;
+    static MSIDKeyOperationUtil *s_keyOperationUtil = nil;
+    dispatch_once(&once, ^{
+        s_keyOperationUtil = [[MSIDKeyOperationUtil alloc] init];
+    });
+    return s_keyOperationUtil;
+}
+
+- (BOOL)isKeyFromSecureEnclave:(SecKeyRef)key
+{
+    if (key)
+    {
+        NSDictionary *attributes = (NSDictionary *)CFBridgingRelease(SecKeyCopyAttributes(key));
+        NSString *attrTokenId = (NSString *)[attributes valueForKey:(NSString *)CFBridgingRelease(kSecAttrTokenID)];
+        NSString *secureEnclaveId = (NSString *)(CFBridgingRelease(kSecAttrTokenIDSecureEnclave));  // id = com.apple.settoken for key in secure enclave
```

Comment:
Same here, you don't need CFBridgingRelease for constants that you didn't allocate

Reply:
Updated